### PR TITLE
Base64 encode statusDesc to fix errors sent from server.

### PR DESCRIPTION
**Pull request by peter-edge**
Signed-off-by: Peter Edge <peter.edge@gmail.com>

Fixes https://github.com/grpc/grpc-go/issues/576, https://github.com/grpc/grpc-go/issues/606

### CODE REVIEW

File: transport/http2_server.go [hunk 1]
```diff
@@ -485,7 +486,11 @@ func (t *http2Server) WriteStatus(s *Stream, statusCode codes.Code, statusDesc s
 			Name:  "grpc-status",
 			Value: strconv.Itoa(int(statusCode)),
 		})
-	t.hEnc.WriteField(hpack.HeaderField{Name: "grpc-message", Value: statusDesc})
+	value := ""
+	if statusDesc != "" {
+		value = base64.StdEncoding.EncodeToString([]byte(statusDesc))
```
Comment (bradfitz) [inline transport/http2_server.go]: base64 should things should have keys ending in `-bin` per http://www.grpc.io/docs/guides/wire.html

Comment (peter-edge) [inline transport/http2_server.go]: Fixed

**Comment by iamqizhao** [comment]
This does not interop with other languages which have no idea what the grpc-message-bin mean.

**Comment by iamqizhao** [comment]
you probably should file issues to grpc/grpc and grpc/grpc-java and have your proposal there. Once we reach agreement, we can check in the fix at same time to minimize potential breakage.

**Comment by bradfitz** [comment]
What do the other languages do if the message contains bogus bytes?

**Comment by peter-edge** [comment]
@bradfitz I'm gonna file an issue over on grpc/grpc once I do some investigation into grpc/grpc and grpc-java and I'll cc you when I get there. I'm context switching like a maniac today so let me pop some stuff off my stack first

**Comment by bradfitz** [comment]
@peter-edge, thanks. I look forward to it.

**Comment by iamqizhao** [comment]
I think https://github.com/grpc/grpc/issues/4672 is the decision.

**Comment by peter-edge** [comment]
Well shoot, it looks like the discussion is over, looks like they are doing this already in grpc-java https://github.com/grpc/grpc-java/pull/1517/files so I don't think it's worth it to discuss base64 vs percent encoding, right?

So if the decision has already effectively been made, is it fair for us to just:

- Keep it as `grpc-message`
- Percent encode/decode the message?

**Comment by peter-edge** [comment]
I can't say I'm the biggest fan of it, I'd personally prefer base64, but @bradfitz I would argue it's not worth going against the tide here right?

**Comment by peter-edge** [comment]
I did this https://github.com/grpc/grpc-go/pull/610

**Pull request closed without merging**